### PR TITLE
Remove for each()

### PR DESCRIPTION
**Pull request by kewisch**

### CODE REVIEW

File: build/ical.js [hunk 1]
```diff
@@ -3304,9 +3317,11 @@ var ICAL = ICAL || {};
 
       // We need our own copy with a few keys set
       var parts = {};
-      for each(var p in ["BYDAY", "BYWEEKNO", "BYMONTHDAY", "BYMONTH", "BYYEARDAY"]) {
-        if(p in this.rule.parts) {
-          parts[p] = this.rule.parts[p];
+      const rules = ["BYDAY", "BYWEEKNO", "BYMONTHDAY", "BYMONTH", "BYYEARDAY"];
```
Comment (lightsofapollo) [inline build/ical.js:3320]: Any reason to use `const` here over var? const is not implemented everywhere and its inside a function so its not really what I would think of as `const`

Comment (kewisch) [inline build/ical.js:3320]: No special reason. I'm fine with changing to var.

File: lib/ical/recur.js [hunk 2]
```diff
@@ -125,8 +130,9 @@ var ICAL = ICAL || {};
       if(this.interval != 1) {
         str += ";INTERVAL=" + this.interval;
       }
-      str += [";" + k + "=" + this.parts[k]
```
Comment (lightsofapollo) [inline lib/ical/recur.js:128]: odd I wonder why It did not throw a syntax error

``` js
[";" + k + "=" + this.parts[k]
```

**Comment by lightsofapollo** [comment]
r=lightsofapollo looks good to me

File: lib/ical/recur.js (hunk 2 above)
Comment (kewisch) [inline lib/ical/recur.js:128]: In newer JS this is actually valid, i.e array comprehension. I bet its not supported everywhere though.

``` js
[ stuff for (stuff in someArray) ]
```

**Pull request merged**